### DISCUSSION TITLE
should work with OpenAI-compatible self-hosted endpoints

### DISCUSSION THREAD

**Discussion by 7flash** [langfuse/langfuse#1004] [Ideas]
Currently I can switch to langchain when using OpenAI

```
- import openai
+ from langfuse.openai import openai
```

But in case if I am hosting my own model with openai-compatible API using vllm or litellm, I would like still to benefit from langfuse observability

```
os.environ["OPENAI_ENDPOINT"] = ""
```

**Comment by Manouchehri** [comment]
If you're using LiteLLM, then this should already work. We're using it already. https://docs.litellm.ai/docs/proxy/logging#logging-proxy-inputoutput---langfuse

**Comment by marcklingen** [comment]
Closing this thread as this works already when changing the base url of the sdk.
